Structural/Decorator/JsonRenderer.php

```diff
@@ -8,4 +8,16 @@ public function renderData(): string
     {
         return json_encode($this->wrapped->renderData());
     }
+
+    /**
+     * Renders data as JSON based on options passed.
+     * In terms of decorator patters, it represents additional extended behaviour.
+     *
+     * @param int $options Options to be used to encode data into JSON.
+     * @return string Rendered output
+     */
+    public function renderDataWithOptions($options = 0): string
```

Comment:
Sorry @andrewnester probably I didn't explain clearly. Maybe a code example will be better. 
To me the pull request is good as is, but I think It could be better this way.

```php
class JsonRenderer extends RendererDecorator
{
  private $options;

  public  __construct($options = 0)
  {
    $this->options = $options;
  }

  public function setOptions($options)
  {
     $this->options = $options;
  }

  public function renderData(): string
  {
     return json_encode($this->wrapped->renderData(), $options);
  }
}

// Example
$service = new Decorator\Webservice('foo/bar');
$decorated_service1 = new Decorator\JsonRenderer($this->service, JSON_UNESCAPED_SLASHES);
$decorated_service2 = new FooRenderer($decorated_service1);
$rendered_data = $decorated_service2->renderData();
```